climada/util/plot.py

```diff
@@ -472,3 +472,70 @@ def get_transformation(crs_in):
     except KeyError:
         units = '°'
     return crs_epsg, units
+
+
+def bar_plot(ax, data, colors=None, total_width=0.8, single_width=1, legend=True):
+    """Draws a bar plot with multiple bars per data point.
+    https://stackoverflow.com/questions/14270391/python-matplotlib-multiple-bars
+
+    Parameters
+    ----------
+    ax : matplotlib.pyplot.axis
+        The axis we want to draw our plot on.
+
+    data: dictionary
+        A dictionary containing the data we want to plot. Keys are the names of the
+        data, the items is a list of the values.
+
```

Comment:
Could this be extended to work with any kind of iterable?
I'd guess normally one would already have a set of variables such as var1 = [1,2,3], var2 = [1,2,3], etc.
Not that it's hard to convert (var1, var2, var3) into a dictionary, but would be neat if this needn't be done by one's-self

Reply:
I am confused here. The idea is to do bar plots where for each xticks there are multiple grouped bars. To have clear labeling, the sub-elements of each group are passed as dictionnary items.

Reply:
i mean, usually you probably don't have a dictionary, but a set of pre-defined variables. Of course they could first be converted to a dictionary, and then passed into the function. But maybe the function could do this internally, so the user can either provide such a dictionary, or a set of variables.

Also, the variable names or keys in the current case are just relevant for labelling the legend. There is no option at the moment to pass group-labels (aka the xticks).

Reply:
Well, you mean a two-dimensional list or np.array? But then how to make the legends if you do not have the variable series names?

The xticks were just added (see last commit).

Reply:
xticks: yep, i see. 

no, i just mean the case where you dont have a dictionary {"var1" = [1,2,3]; "var2" = [1,2,3]}, but a set of variables (var1 = [1,2,3], var2 = [1,2,3]) which you already defined somewhere on the way. instead of taking your variables and converting them manually to a dictionary first, maybe the function could check for the input types (e.g. is it already a dict --> all good, is it a set of variables --> make one out of those with keys and values)

Reply:
Well... I still do not understand, but more convenience is certainly worth it. 

So, what you propose is to allow the user to give dictionnary or any other nested iterable? In your example you replaced the dictionary by a tuple of list? So possible entries would be:

`{'var1': [1,2,3], 'var2':[1,2,3]}`
`([1,2,3], [1,2,3])`
`[[1,2,3], [1,2,3]]`

Or would you like to have an arbitrary number of keyword arguments?

`multibar(var1=[1,2,3], var2=[1,2,3], var3=[1,2,3], ticklabel=['a', 'b', 'c']`

I am not sure how I could implement the latter together with the dictionary, which in my opinion is cleaner.

Reply:
Ideally, i would like both this to work:
`multibar({'var1': [1,2,3], 'var2':[1,2,3]}, ticklabel=['a', 'b', 'c'])`
and this:
`multibar((var1, var2), ticklabel=['a', 'b', 'c'])` (where var1 and var2 have been defined previously somewhere).

The latter one (`multibar(var1=[1,2,3], var2=[1,2,3], var3=[1,2,3], ticklabel=['a', 'b', 'c'])` with an arbitrary amount of keyword arguments would also be possible, but is probably over-engineered (cf. (https://treyhunner.com/2018/04/keyword-arguments-in-python/#Capturing_arbitrary_keyword_arguments) )

Reply:
Well, so you are proposing to have lists of lists, tuples of lists, or whatever 2d arrays. This is fine, but then there would be no legend, which in my opinion makes the graphic not readable. In your example, one should simply write

`multibar({'var1': var1, 'var2': var2}, ticklabel=['a','b','c'])`

which in my opinion is not more complex than 

`multibar((var1, var2), ticklabel=['a', 'b', 'c'])`.

Reply:
What do you think? 

Also, the multi keyword argument route can of course be done, but together with the dictionary, it seems confusing to me (both technically and user side).

Reply:
Right. I wouldn't do the dictionary _and_ the multi keyword argument. that's indeed to confusing.
I guess you're right, in case one already has variables, it's probably easy enough to manually make a dict via `multibar({'var1': var1, 'var2': var2}, ticklabel=['a','b','c'])`

So let's leave it as is

Reply:
Very well, thanks a lot for the great review :D !